nulib/meadow
# Use a nodejs v16.x compatible require-hook in local lambdas

# Summary 

Allow devs to use NodeJS v16.x locally by replacing the `require-hook` dependency

# Specific Changes in this PR
- Replace [`require-hook`](https://www.npmjs.com/package/require-hook) with [`@contrast/require-hook`](https://www.npmjs.com/package/@contrast/require-hook) in the lambda shim
- Modify the `aws-sdk` intercept/patching code to work with `@contrast/require-hook`

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test

1. `mix assets.install`
2. Run some ingests locally
3. Make sure all the lambdas that access S3 objects work OK
4. Search local console log output for `Patching aws-sdk v2.x.x`

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes


# Tested/Verified
